tsdb/testutil.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tsdb
+
+import (
+	"testing"
+
+	prom_testutil "github.com/prometheus/client_golang/prometheus/testutil"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb/chunks"
+)
+
+const (
+	float = "float"
+)
+
+type tsValue struct {
+	Ts                 int64
+	V                  int64
+	CounterResetHeader histogram.CounterResetHint
+}
+
+type sampleTypeScenario struct {
+	sampleType string
+	appendFunc func(appender storage.Appender, lbls labels.Labels, ts, value int64) (storage.SeriesRef, sample, error)
+	sampleFunc func(ts, value int64) sample
+}
+
+// TODO: native histogram sample types will be added as part of out-of-order native histogram support; see #11220.
+var sampleTypeScenarios = map[string]sampleTypeScenario{
+	float: {
+		sampleType: sampleMetricTypeFloat,
+		appendFunc: func(appender storage.Appender, lbls labels.Labels, ts, value int64) (storage.SeriesRef, sample, error) {
+			s := sample{t: ts, f: float64(value)}
+			ref, err := appender.Append(0, lbls, ts, s.f)
+			return ref, s, err
+		},
+		sampleFunc: func(ts, value int64) sample {
+			return sample{t: ts, f: float64(value)}
+		},
+	},
+	// intHistogram: {
```

Comment:
Why is this commented out? Tests don't work yet in this PR?

I see how this PR is refactoring the test structure for all the different types of samples. Just trying to understand if this can be uncommented now or we have to wait for other code to be merged.

Reply:
We need to wait for the other code for handling OOO native histogram samples to be merged first